WordPress/src/main/java/org/wordpress/android/ui/debug/DebugSettingsAdapter.kt

```diff
@@ -0,0 +1,58 @@
+package org.wordpress.android.ui.debug
+
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView.Adapter
+import org.wordpress.android.ui.debug.DebugSettingsItemViewHolder.ButtonViewHolder
+import org.wordpress.android.ui.debug.DebugSettingsItemViewHolder.FeatureViewHolder
+import org.wordpress.android.ui.debug.DebugSettingsItemViewHolder.HeaderViewHolder
+import org.wordpress.android.ui.debug.DebugSettingsItemViewHolder.RowViewHolder
+import org.wordpress.android.ui.debug.DebugSettingsViewModel.UiItem
+import org.wordpress.android.ui.debug.DebugSettingsViewModel.UiItem.Button
+import org.wordpress.android.ui.debug.DebugSettingsViewModel.UiItem.Feature
+import org.wordpress.android.ui.debug.DebugSettingsViewModel.UiItem.Header
+import org.wordpress.android.ui.debug.DebugSettingsViewModel.UiItem.Row
+import org.wordpress.android.ui.debug.DebugSettingsViewModel.UiItem.Type
+import org.wordpress.android.ui.debug.DebugSettingsViewModel.UiItem.Type.BUTTON
+import org.wordpress.android.ui.debug.DebugSettingsViewModel.UiItem.Type.FEATURE
+import org.wordpress.android.ui.debug.DebugSettingsViewModel.UiItem.Type.HEADER
+import org.wordpress.android.ui.debug.DebugSettingsViewModel.UiItem.Type.ROW
+
+class DebugSettingsAdapter : Adapter<DebugSettingsItemViewHolder>() {
```

Comment:
If you could [ListAdapter](https://developer.android.com/reference/kotlin/androidx/recyclerview/widget/ListAdapter) here then line 22 and lines 24 - 33 and line 48 can be removed

Reply:
Good call! I initially refrained from making any changes to these classes to prevent the commit history from getting messed up, but I think in this case it's worth it. Updated in f428f4b6bd1ed0bd7d788d4e8576e530c2d1fd64.